packages/machine/test/integration/install-then-set-state.spec.ts

```diff
@@ -0,0 +1,239 @@
+const JEST_TEST_WAIT_TIME = 30000;
+
+// ProxyFactory.createProxy uses assembly `call` so we can't estimate
+// gas needed, so we hard-code this number to ensure the tx completes
+const CREATE_PROXY_AND_SETUP_GAS = 6e9;
+
+// The AppRegistry.setState call _could_ be estimated but we haven't
+// written this test to do that yet
+const SETSTATE_COMMITMENT_GAS = 6e9;
+
+// Also we can't estimate the install commitment gas b/c it uses
+// delegatecall for the conditional transaction
+const INSTALL_COMMITMENT_GAS = 6e9;
+
+let networkId: number;
+let provider: JsonRpcProvider;
+let wallet: Wallet;
+let network: NetworkContext;
+
+// TODO: When we add a second use case of this custom mathcer,
+//       move it and its typigns into somewhere re-usable
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toBeEq(expected: BigNumber): BigNumber;
+    }
+  }
+}
+expect.extend({
+  toBeEq(received: BigNumber, argument: BigNumber) {
+    return {
+      pass: received.eq(argument),
+      message: () => `expected ${received} not to be equal to ${argument}`
+    };
+  }
+});
+
+// TODO: This will be re-used for all integration tests, so
+//       move it somewhere re-usable when we add a new test
+beforeAll(async () => {
+  dotenv.config();
+
+  // Can use ! because dotenv-safe ensures value is set
+  const host = process.env.DEV_GANACHE_HOST!;
+  const port = process.env.DEV_GANACHE_PORT!;
+  const mnemonic = process.env.DEV_GANACHE_MNEMONIC!;
+
+  function pathForAccount(idx: number) {
+    return `m/44'/60'/0'/0/${idx}`;
+  }
+
+  provider = new JsonRpcProvider(`http://${host}:${port}`);
+  wallet = Wallet.fromMnemonic(mnemonic, pathForAccount(1)).connect(provider);
+  networkId = (await provider.getNetwork()).chainId;
+
+  const relevantArtifacts = [
+    AppRegistry,
+    ETHBucket,
+    MultiSend,
+    NonceRegistry,
+    StateChannelTransaction
+  ];
+
+  network = {
+    // Fetches the values from build artifacts of the contracts needed
+    // for this test and sets the ones we don't care about to 0x0
+    ETHBalanceRefund: AddressZero,
+    ...relevantArtifacts.reduce(
+      (accumulator, artifact) => ({
+        ...accumulator,
+        [artifact.contractName]: artifact.networks[networkId].address
+      }),
+      {}
+    )
+  } as NetworkContext;
+});
+
+/**
+ * This test suite tests setting up a channel and then installing a new app into it.
+ * We re-use the ETHBucket App (which is the app ETH Free Balance uses) as the
+ * test app being installed. We then set the values to [1, 1] in that app
+ * and trigger the InstallCommitment on-chain to resolve that app and verify
+ * the balances have been updated on-chain.
+ */
+describe("Scenario: install app, set state, put on-chain", () => {
+  jest.setTimeout(JEST_TEST_WAIT_TIME);
+
+  it("returns the funds the app had locked up", async done => {
+    const signingKeys = [
+      new SigningKey(hexlify(randomBytes(32))),
+      new SigningKey(hexlify(randomBytes(32)))
```

Comment:
Why randomize? I prefer predefined constants.

Reply:
Aiming for unique values without hard-coding arbitrary strings. Ideally `randomBytes` would take a "seed" value.